CoreMetrics/CoreMetrics.cs

```diff
@@ -75,7 +75,7 @@ private JObject GetBlocksTime(uint nBlocks, uint lastHeight)
 
             JArray array = new JArray();
             uint heightToBegin = lastHeight > 0 ? lastHeight - nBlocks : Blockchain.Singleton.Height - nBlocks;
-            for (uint i = heightToBegin; i <= Blockchain.Singleton.HeaderHeight; i++)
+            for (uint i = heightToBegin; i <= heightToBegin + nBlocks; i++)
             {
                 JObject json = new JObject();
                 Header header = Blockchain.Singleton.Store.GetHeader(i);
```

Comment:
We must to check if is null

Reply:
You mean `Blockchain.Singleton.Height`?

Reply:
The header

Reply:
That is true.
I do know when it will be `null` in practice, because we are checking the limit `Blockchain.Singleton.HeaderHeight`.
But looks like to be a protection. Let's add it in another PR if it really can happen.